### ISSUE TITLE
Is it possible to use YOLOE only for detection without segmentation?

### ISSUE THREAD

**Issue by xiaokj37** [ultralytics/ultralytics#20815]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm working on a task where I only need to obtain a series of target box classifications, without any segmentation outputs. 
I want to avoid instance segmentation to reduce the computational cost of the model
The desired behavior is similar to the prompt-free setting with the mask blocked, as described in the documentation (https://docs.ultralytics.com/models/yoloe/).

I've noticed that the yoloe-11.yaml configuration specifies YOLOEDetect as the final layer, which seems suitable for detection-only tasks.

However, when looking for pre-trained weights, I primarily find models with a "-seg" suffix (e.g., yoloe-11s-seg.pt). Are there pre-trained weights available that are optimized or intended specifically for detection-only using configurations like yoloe-11.yaml without the segmentation head? Or is the recommended approach to use the "-seg" models and simply ignore/disable the mask output if only detection is needed?

Any clarification on the best way to achieve efficient detection-only inference and the availability of corresponding pre-trained weights would be greatly appreciated.

Thanks for your excellent work on YOLOE!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xiaokj37, thank you for your interest in Ultralytics and for your detailed question about YOLOE! 🚀 This is an automated response to help you get started—an Ultralytics engineer will also assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/), where you'll find guidance on YOLOE and examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Many common questions about model configuration, detection-only tasks, and weights are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) for us to assist you more efficiently.

For custom training or inference ❓ Questions, please include as much relevant detail as possible—such as config files, dataset samples, or logs—and verify you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for real-time support and discussion:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for live chat
- [Discourse](https://community.ultralytics.com/) for in-depth topics
- [Subreddit](https://reddit.com/r/Ultralytics) for sharing knowledge and experiences

## Upgrade

To ensure the latest features and bug fixes, upgrade to the newest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your question! An Ultralytics engineer will review your inquiry about detection-only inference and pre-trained weights for YOLOE and respond with further guidance soon.

**Comment by xiaokj37** [comment]
I tried this approach, but it seems that the final result is empty detection.
The code and output log are as follows:

Code:
`from ultralytics import YOLO, YOLOE

# Initialize a YOLOE model
# model = YOLOE("pretrain/yoloe-11s-seg.pt").cuda()
model = YOLO("yoloe-11s.yaml").load("pretrain/yoloe-11s-seg.pt")
model.save("pretrain/yoloe-11s.pt")

test_model = YOLO("pretrain/yoloe-11s.pt").cuda()
results = model("ultralytics/assets/bus.jpg", save=True)`

Log
`Transferred 592/592 items from pretrained weights

image 1/1 /root/MobileAIR/yoloe/ultralytics/assets/bus.jpg: 640x640 (no detections), 10.2ms
Speed: 6.7ms preprocess, 10.2ms inference, 23.3ms postprocess per image at shape (1, 3, 640, 640)
Results saved to runs/detect/predict3`

**Comment by Y-T-G** [comment]
Did you set the prompt?

**Comment by Y-T-G** [comment]
You can try this:
```python
model = YOLO("yoloe-11s.yaml").load("yoloe-11s-seg.pt")
model.eval()                                                           model.ckpt = {}
model.save("yoloe-11s.pt")                                             model = YOLO("yoloe-11s.pt", task="detect")
names = ["person", "bus"]                                              model.set_classes(names, model.get_text_pe(names))
results = model()
```
